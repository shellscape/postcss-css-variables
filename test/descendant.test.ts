import { fixture, process } from './helpers';

const tests = {
  'should work with css4 descendant selector type "nesting"': 'css4-descendant-selector',
  'should work with descendant selector type "nesting"': 'descendant-selector',
  'should work with direct descendant selector': 'direct-descendant-selector',
  'should work with direct descendant selector where variables are scoped in a descendant selector':
    'direct-descendant-selector-descendant-scope',
  'should work with direct descendant selector where variables are scoped in a direct descendant selector':
    'direct-descendant-selector-direct-descendant-scope'
};

describe('descendant', async () => {
  test.each(Object.entries(tests))('%s', async (_desc, file) => {
    const css = await fixture(`descendant/${file}`);
    const result = await process(css);
    expect(result).toMatchSnapshot();
  });
});
