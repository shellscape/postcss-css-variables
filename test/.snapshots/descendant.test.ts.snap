// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`descendant > should work with css4 descendant selector type "nesting" 1`] = `".some-foo-group >> .box-foo{width:180px}.some-bar-group >> .box-bar{width:300px}.some-foo-group{width:150px;.box-foo{width:180px}}"`;

exports[`descendant > should work with descendant selector type "nesting" 1`] = `".some-foo-group .box-foo{width:150px}.some-bar-group .box-bar{width:300px}"`;

exports[`descendant > should work with direct descendant selector 1`] = `".foo > .bar{width:150px}"`;

exports[`descendant > should work with direct descendant selector where variables are scoped in a descendant selector 1`] = `".baz > .qux > .wan{width:300px}.some-parent > .baz > .qux > .wan{width:300px}.some-parent > .baz > .wan > .qux{width:300px}.some-parent > .baz > .qux > .wan .further-child .ger{width:320px}"`;

exports[`descendant > should work with direct descendant selector where variables are scoped in a direct descendant selector 1`] = `".foo > .bar > .baz{width:400px}.qux + .foo > .bar > .baz{width:400px}.foo > .bar + .baz{width:undefined}"`;
